### ISSUE TITLE
capture audit log for each notification

### ISSUE THREAD

**Issue by sjain** [excid3/noticed#4]
Would be nice to capture:

1. which notification(s) were sent and when.
2. which notification was not sent (which condition triggered the decision not to send).

**Comment by excid3** [comment]
We can probably add some callbacks you can hook into for this.

**Comment by excid3** [comment]
Started implementing this tonight. I have callbacks running but they'll need some fine-tuning to get right.

Right now I'm looking at callbacks for:

* [x] On the notification around the delivery
* [x] Inside the delivery methods themselves
* [ ] On the notification for each delivery method

This will be flexible so you can write audit code either on the notification or inside each of the delivery methods. Either option would work.

I believe returning `false` in the `before` callbacks should also halt processing which would match ActiveModel's callbacks. We're using the ActiveModel::Callbacks module so I would assume it matches.

**Comment by excid3** [comment]
I decided we'd add callbacks for each delivery method as they're queued. This won't run them as they're actually executed in the background job, but you'll at least know it was queued to run.

To have a callback that runs around the delivery in the background job, you can simply create your own custom delivery method and use that.

```ruby
class TwilioWithAuditing < Noticed::DeliveryMethods::Twilio
  after_delivery do
    # audit code
  end
end
```

**Comment by excid3** [comment]
Alrighty, released this as v1.1! :shipit:
